src/rdf4cpp/rdf/storage/node/LiteralBackend.cpp

```diff
@@ -0,0 +1,19 @@
+#include "LiteralBackend.hpp"
+#include <tuple>
+namespace rdf4cpp::rdf::storage::node {
+
+LiteralBackend::LiteralBackend(std::string lexical, const NodeID &dataType, std::string langTag)
+    : datatype_id_(dataType),
+      lexical(std::move(lexical)),
+      lang_tag(std::move(langTag)) {}
+std::strong_ordering LiteralBackend::operator<=>(const LiteralBackend &other) const noexcept {
+    return std::tie(this->datatype_id_.node_id(), this->lexical, this->lang_tag) <=> std::tie(other.datatype_id_.node_id(), other.lexical, other.lang_tag);
+}
+
+std::string LiteralBackend::quote_lexical() const {
+    return "\"" + lexical + "\"";
```

Comment:
this is problematic if lexical contains an unescaped "

Reply:
agreed. I would add an todo for now. I need to re-check specs on this. If we escape `"` here I would also escape everything else which is not allowed in N-Tripels/N-Quads (if there is anything else). I expect this function to be used by those serialisations, mainly.